x-pack/plugin/sql/src/main/antlr/SqlBase.g4

```diff
@@ -454,7 +454,7 @@ DIGIT_IDENTIFIER
     ;
 
 TABLE_IDENTIFIER
-    : (LETTER | DIGIT | '_' | '@' | ASTERISK)+
+    : (LETTER | DIGIT | '_' | '@')+
```

Comment:
For consistency I would remove `@` so only letters, digits and `_` are allowed - going forward I think it's best to use quote for pretty much all non-alphabetical indices to avoid ambiguities.